core/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -490,8 +490,10 @@ protected void parseCreateField(ParseContext context, List<IndexableField> field
 
     @Override
     protected void doMerge(Mapper mergeWith, boolean updateAllTypes) {
+        final DateFieldMapper other = (DateFieldMapper) mergeWith;
+        if (!this.fieldType().equals(other.fieldType()))
+            throw new IllegalArgumentException("date field cannot be updated");
```

Comment:
Thx for your comments, I modified the error message and specified it's _date field's format_ which cannot be updated. And the `fieldType()`'s [`equals()`](https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java#L193-L199) method has been re-implemented to just compare the format of a Date, isn't it ?